.pre-commit-config.yaml

```diff
@@ -7,45 +7,45 @@ repos:
     hooks:
       - id: black
         language_version: python3
-        args: ['--target-version', 'py37']
-  - repo: https://github.com/timothycrosley/isort
-    # isort config is in setup.cfg
-    rev: 5.6.4
+        args: ['--target-version', 'py38']
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: 'v0.0.290'
     hooks:
-      - id: isort
-  - repo: https://github.com/pycqa/flake8
-    # flake8 config is in setup.cfg
-    rev: 3.8.4
-    hooks:
-      - id: flake8
-        additional_dependencies:
-          - flake8-comprehensions
-          - flake8-assertive
-          - flake8-print
-  - repo: https://github.com/rtts/djhtml
-    rev: v1.4.13
-    hooks:
-      - id: djhtml
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
+
   - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: v2.5.1
+    rev: v3.0.3
     hooks:
       - id: prettier
-        types_or: [css, javascript, json, yaml]
+        types_or: [css, scss, javascript, ts, tsx, json, yaml]
+        additional_dependencies:
+          # Keep in sync with package.json
+          - prettier@3.0.1
   - repo: https://github.com/pre-commit/mirrors-eslint
-    rev: v8.8.0
+    rev: v8.49.0
     hooks:
       - id: eslint
         types: [file]
-        files: \.(js)$
+        files: \.(js|ts|tsx)$
         args: [--report-unused-disable-directives]
         additional_dependencies:
-          - eslint@8.8.0
+          # Keep in sync with package.json
+          - eslint@8.46
+          - '@typescript-eslint/eslint-plugin@6.2.1'
+          - '@typescript-eslint/parser@6.2.1'
+          - '@wagtail/eslint-config-wagtail@0.4.0'
   - repo: https://github.com/thibaudcolas/pre-commit-stylelint
-    rev: v14.2.0
+    rev: v15.10.3
     hooks:
       - id: stylelint
-        files: \.css$
+        files: \.scss$
```

Comment:
I think we only use CSS in this project.

Reply:
No copy/paste job shall go unpunished ;)